### PR TITLE
Create scikit-learn-intelex package

### PR DESCRIPTION
**Pull request by Pahandrovich**

### CODE REVIEW

File: daal4py/sklearn/monkeypatch/dispatcher.py [hunk 1]
```diff
@@ -145,6 +145,10 @@ def enable(name=None, verbose=True):
     else:
         for key in _get_map_of_algorithms():
             do_patch(key)
+    if deprecation and sys.stderr is not None:
+        sys.stderr.write(
+            "daal4py patching will be deprecated in the future, "
+            "use iskex module instead\n")
```
Comment (SmirnovEgorRu) [inline daal4py/sklearn/monkeypatch/dispatcher.py:151]: We will need to work on messaging, I prefer to include full name "Intel(R) Extension for Scikit-learn" and provide info how to get it (like pip install intel-sklearn-extension) and how to enable patching.

+ As I remember python has native instruments for deprecation warnings, not just write to stderr.

Comment (Pahandrovich) [reply daal4py/sklearn/monkeypatch/dispatcher.py:151]: Thanks!
What about 
`daal4py patching will be deprecated in the future, "
            "use Intel(R) Extension for Scikit-learn module instead\n"
            "pip install intel-sklearn-extension\n"
            "python -m iskex <your_script>\n"
            "or\n"
            "from iskex import patch_sklearn\n"
            "patch_sklearn()\n"`
User can disable this warning with patch_sklearn(deprecation=False)

Comment (SmirnovEgorRu) [reply daal4py/sklearn/monkeypatch/dispatcher.py:151]:
```
Scikit-learn patching with daal4py is deprecated and will be removed in the future.
Please, use Intel(R) Extension for Scikit-learn module instead (pip install intel-sklearn-extension)
To enable patching, please, use one of options:
1) python -m iskex <your_script>
2) from iskex import patch_sklearn
   patch_sklearn()
```

Also, please keep in mind, that names of packages are not finalized, so potentially they can be changed.

File: daal4py/sklearn/monkeypatch/dispatcher.py [hunk 2]
```diff
@@ -145,6 +146,16 @@ def enable(name=None, verbose=True):
...
+    if deprecation:
+        warn("Scikit-learn patching with daal4py is deprecated "
+            "and will be removed in the future.\n"
+            "Please, use Intel(R) Extension for Scikit-learn module instead "
+            "(pip install intel-sklearn-extension)\n"
+            "To enable patching, please, use one of options:\n"
+            "1) python -m iskex <your_script>\n"
+            "2) from iskex import patch_sklearn\n"
+            "   patch_sklearn()", 
+             FutureWarning, stacklevel=2)
```
Comment (SmirnovEgorRu) [inline daal4py/sklearn/monkeypatch/dispatcher.py:158]: ```DeprecationWarning``` probably?

Comment (Pahandrovich) [reply daal4py/sklearn/monkeypatch/dispatcher.py:158]: DeprecationWarning are intended for other Python developers, and FutureWarning for users
https://docs.python.org/3/library/warnings.html
For example, scikit-learn using FutureWarning
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/cluster/_kmeans.py#L787

File: .circleci/config.yml [hunk 3]
```diff
@@ -22,6 +22,12 @@ jobs:
...
             python setup.py install --single-version-externally-managed --record=record.txt
+      - run:
+          name: Building sklearnex
+          command: |
+            . ~/miniconda/etc/profile.d/conda.sh
+            conda activate bld
+            python setup_sklearnex.py install --single-version-externally-managed --record=record1.txt
 
       - run:
           name: Testing sklearn patches
```
Comment (PetrovKP) [inline .circleci/config.yml:33]: I think it can delete. Leave only sklearnex

File: README.md [hunk 4]
```diff
@@ -1,11 +1,10 @@
...
 [![Join the community on GitHub Discussions](https://badgen.net/badge/join%20the%20discussion/on%20github/black?icon=github)](https://github.com/IntelPython/daal4py/discussions)
 [![PyPI Version](https://img.shields.io/pypi/v/daal4py)](https://pypi.org/project/daal4py/)
-[![Conda Version](https://img.shields.io/conda/vn/conda-forge/daal4py)](https://anaconda.org/conda-forge/daal4py)
 
 
-A simplified API to Intel(R) oneAPI Data Analytics Library that allows for fast usage of the framework suited for Data Scientists or Machine Learning users.  Built to help provide an abstraction to Intel(R) oneAPI Data Analytics Library for either direct usage or integration into one's own framework and extending this beyond by providing drop-in paching for scikit-learn.
+Intel(R) Extension for scikit-learn extending scikit-learn beyond by providing drop-in paching.  Acceleration is achieved through the use of the library Intel(R) oneAPI Data Analytics Library that allows for fast usage of the framework suited for Data Scientists or Machine Learning users.
 
 Running full scikit-learn test suite with daal4py optimization patches:
```
Comment (PetrovKP) [inline README.md:9]:
```suggestion
Running full scikit-learn test suite with scikit-learn-intelex optimization patches:
```

File: README.md [hunk 5]
```diff
@@ -122,17 +107,17 @@ with sycl_context("gpu"):
...
 | *Technical details: float type: float64; HW: Intel(R) Xeon(R) Platinum 8280 CPU @ 2.70GHz, 2 sockets, 28 cores per socket; SW: scikit-learn 0.23.1, Intel® oneDAl (2021.1 Beta 10)* |
 
-daal4py patching affects performance of specific Scikit-learn functionality listed below. In cases when unsupported parameters are used, daal4py fallbacks into stock Scikit-learn. These limitations described below. If the patching does not cover your scenarios, [submit an issue on GitHub](https://github.com/IntelPython/daal4py/issues).
+scikit-learn-intelex patching affects performance of specific Scikit-learn functionality listed below. In cases when unsupported parameters are used, scikit-learn-intelex fallbacks into stock Scikit-learn. These limitations described below. If the patching does not cover your scenarios, [submit an issue on GitHub](https://github.com/IntelPython/daal4py/issues).
 
-⚠️ We support optimizations for the last four versions of scikit-learn. The latest release of daal4py-2021.1 supports scikit-learn 0.21.X,
+⚠️ We support optimizations for the last four versions of scikit-learn. The latest release of scikit-learn-intelex-2021.1 supports scikit-learn 0.21.X,
```
Comment (PetrovKP) [inline README.md:117]:
```suggestion
⚠️ We support optimizations for the last four versions of scikit-learn. The latest release of scikit-learn-intelex-2021.2 supports scikit-learn 0.21.X,
```

File: daal4py/sklearn/_utils.py [hunk 6]
```diff
@@ -21,6 +21,29 @@
...
+            logging.basicConfig(
+                stream=sys.stdout,
+                format='%(levelname)s: %(message)s', level=logLevel.upper())
+            if deprecation:
+                warnings.warn("IDP_SKLEARN_VERBOSE fuctionality is deprecated "
+                              "and will be removed in the future.\n"
+                              "Please, use SKLEARNEX_VERBOSE variable in "
+                              "Intel(R) Extension for Scikit-learn module instead",
+                              FutureWarning, stacklevel=2)
+    except:
```
Comment (PetrovKP) [inline daal4py/sklearn/_utils.py:41]: Apply comment codefactor

File: daal4py/sklearn/monkeypatch/dispatcher.py [hunk 7]
```diff
@@ -145,6 +146,17 @@ def enable(name=None, verbose=True):
     else:
         for key in _get_map_of_algorithms():
             do_patch(key)
+    set_idp_sklearn_verbose(skl_verbose)
```
Comment (PetrovKP) [inline daal4py/sklearn/monkeypatch/dispatcher.py:149]:
```suggestion
    set_idp_sklearn_verbose(deprecation)
```

File: setup_sklearnex.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+          "Programming Language :: Python :: 3.6",
+          "Programming Language :: Python :: 3.7",
+          "Programming Language :: Python :: 3.8",
+          "Programming Language :: Python :: 3.9",
+          "Topic :: Scientific/Engineering",
+          "Topic :: System",
+          "Topic :: Software Development",
+      ],
+      python_requires=">=3.6",
+      install_requires="daal4py==2021.1",
```
Comment (PetrovKP) [inline setup_sklearnex.py:71]: for now
```suggestion
      install_requires="daal4py>=2021.1",
```

File: sklearnex/__init__.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+__all__ = [
+    "patch_sklearn", "unpatch_sklearn", "sklearn_patch_names",
+]
+
+import logging
+import warnings
+import os
+import sys
+logLevel = os.environ.get("SKLEARNEX_VERBOSE")
```
Comment (PetrovKP) [inline sklearnex/__init__.py:30]: Align with IDP_SKLEAN_VERBOSE - move to separate global function

File: sklearnex/dispatcher.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+def patch_sklearn(name=None, verbose=True):
+    patch_sklearn_orig(name, verbose, deprecation=False, skl_verbose=False)
+
+
+def unpatch_sklearn(name=None):
+    unpatch_sklearn_orig(name)
+
+
+def sklearn_patch_names():
```
Comment (PetrovKP) [inline sklearnex/dispatcher.py:31]: bad name, maybe?
```suggestion
def get_patch_names():

```

File: setup_sklearnex.py [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+#===============================================================================
+
+# System imports
+import os
+import sys
+import time
+from setuptools import setup
+from os.path import join as jp
+
+sklearnex_version = (os.environ["SKLEARNEX_VERSION"] if "SKLEARNEX_VERSION" in os.environ
```
Comment (PetrovKP) [inline setup_sklearnex.py:25]:
```suggestion
sklearnex_version = (os.environ["SKLEARN_EXTENSION_VERBOSE"] if "SKLEARN_EXTENSION_VERBOSE" in os.environ
```

**Review by PetrovKP — APPROVED** [review]
I think we can merge as part1

**Comment by PetrovKP** [comment]
@Mergifyio backport rls/2021.2-rls

File: setup_sklearnex.py (hunk 11 above)
Comment (Pahandrovich) [reply setup_sklearnex.py:25]: This is variable for version of package

Comment (PetrovKP) [reply setup_sklearnex.py:25]: Sorry)

**Comment by PetrovKP** [comment]
@Mergifyio backport rls/2021.2-rls

**Pull request merged**